abi/BUILD

```diff
@@ -0,0 +1,134 @@
+# Standard constraint_setting and constraint_values to be used in platforms.
+licenses(["notice"])
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+)
+
+# To add a new constraint_value see https://github.com/bazelbuild/platforms.
+constraint_setting(name = "abi")
+
+constraint_value(
+    name = "atmega328",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "eabi",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "eabihf",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "elf",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnu",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnu_ilp32",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnueabi",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnuabi64",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnueabihf",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnuspe",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnux32",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "linuxkernel",
```

Comment:
This doesn't seem like it belongs? Kernel code can be built with various other options from this list, there's nothing particularly special about object files built for the kernel.

Reply:
Fair. This was just a value listed in https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-3 so it was included in what I added. As per most of my comments on this PR, I'm happy to remove this and have a more focused conversation on this.

Reply:
I don't think using Rust, or Go, or linux kernel builds, or whatever as the pattern to emulate wins.

Was there a design proposal for this? It is much easier to discuss in a doc than in PR review.

Reply:
Aside from [this](https://docs.google.com/document/d/1CgU-GKocMAfsUSI3bbGZ0YRkOWczitIoKs29x3zR914/edit#heading=h.3fbh1otqm5sz) there's just the github issue (https://github.com/bazelbuild/platforms/issues/38). I tried emailing //bazel-dev@googlegroups.com per the README but the address no longer seems valid.